koku/api/settings/tag_mappings/view.py

```diff
@@ -0,0 +1,176 @@
+
+        return response
+
+
+class SettingsTagMappingParentView(generics.GenericAPIView):
+    queryset = EnabledTagKeys.objects.exclude(child__parent__isnull=False).filter(enabled=True)
+    serializer_class = EnabledTagKeysSerializer
+    permission_classes = (SettingsAccessPermission,)
+    filter_backends = (DjangoFilterBackend,)
+    filterset_class = SettingsEnabledTagKeysFilter
+
+    @method_decorator(never_cache)
+    def get(self, request: Request, **kwargs):
+        filtered_qset = self.filter_queryset(self.get_queryset())
+        serializer = self.serializer_class(filtered_qset, many=True)
+        paginator = ListPaginator(serializer.data, request)
+        response = paginator.paginated_response
+
+        return response
+
+
+class SettingsTagMappingChildAddView(APIView):
+    permission_classes = (SettingsAccessPermission,)
+    serializer_class = TagMappingSerializer
+
+    @transaction.atomic
+    def put(self, request):
+        try:
+            parent_uuid = request.data.get("parent")
+            children_uuids = request.data.get("children", [])
+
+            parent = EnabledTagKeys.objects.get(uuid=parent_uuid, enabled=True)
+            children = EnabledTagKeys.objects.filter(uuid__in=children_uuids, enabled=True)
+
+            if not parent or not children.exists():
+                return Response({"detail": "Invalid parent or children UUIDs."}, status=status.HTTP_400_BAD_REQUEST)
+
+            # Check if any of the children_uuids are already in TagMapping
+            if TagMapping.objects.filter(child__uuid__in=children_uuids).exists():
+                return Response(
+                    {"detail": "Some child UUIDs are already inserted in TagMapping."},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+
+            # Validate if parent can be a child in any existing mappings
+            if TagMapping.objects.filter(child=parent_uuid).exists():
+                return Response({"detail": "A child can't become a parent."}, status=status.HTTP_400_BAD_REQUEST)
+
+            # Validate if children can be parents in any existing mappings
+            if TagMapping.objects.filter(parent__in=children_uuids).exists():
+                return Response({"detail": "A parent can't become a child."}, status=status.HTTP_400_BAD_REQUEST)
+
+            with transaction.atomic():
+                # Create TagMapping for each child
+                for child in children:
+                    TagMapping.objects.create(parent=parent, child=child)
+
+            # Serialize and return the created TagMapping
+            serializer = TagMappingSerializer(TagMapping.objects.filter(parent=parent), many=True)
+
+            return Response(serializer.data, status=status.HTTP_203_NON_AUTHORITATIVE_INFORMATION)
+
+        except EnabledTagKeys.DoesNotExist:
+            return Response({"detail": "Invalid parent or children UUIDs."}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
This whole try/except is questionable. What part of this flow actually raises the `EnabledTagKeys.DoesNotExist`? That's the part of the code that should be wrapped with the try/except, not this entire function.

Reply:
He was using it weed out invalid uuids, I just update the serializer to check that all uuids are valid. Removing the need for the try except all together.